File: Models/Lease.cs
namespace PropertySphere.Models;

public class Lease
{
    public string? Id { get; set; }
    public string UnitId { get; set; } = string.Empty;
    public List<string> RenterIds { get; set; } = new();
    public decimal LeaseAmount { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string? LegalDocumentId { get; set; }

    // Utility pricing per unit (e.g., per kWh for power, per gallon for water)
    public decimal PowerUnitPrice { get; set; } = 0.12m; // default $0.12 per kWh
    public decimal WaterUnitPrice { get; set; } = 0.005m; // default $0.005 per gallon

    public bool IsActive => DateTime.Today >= StartDate && DateTime.Today <= EndDate;
}

File: Models/lease.py
from dataclasses import dataclass, field
from typing import Optional
from datetime import datetime, date
from decimal import Decimal


@dataclass
class Lease:
    UnitId: str = ""
    RenterIds: list[str] = field(default_factory=list)
    LeaseAmount: Decimal = Decimal("0.00")
    StartDate: datetime = field(default_factory=datetime.now)
    EndDate: datetime = field(default_factory=datetime.now)
    LegalDocumentId: Optional[str] = None
    PowerUnitPrice: Decimal = Decimal("0.12")  # default $0.12 per kWh
    WaterUnitPrice: Decimal = Decimal("0.005")  # default $0.005 per gallon
    Id: Optional[str] = None
    
    @property
    def IsActive(self) -> bool:
        today = date.today()
        return self.StartDate.date() <= today <= self.EndDate.date()
    
    @classmethod
    def collection_name(cls) -> str:
        return "Leases"
